x/distribution/types/params.go

```diff
@@ -67,6 +75,11 @@ func (p Params) ValidateBasic() error {
 			"community tax should non-negative and less than one: %s", p.CommunityTax,
 		)
 	}
+	if p.SecretFoundationTax.IsNegative() || p.SecretFoundationTax.GT(sdk.OneDec()) {
+		return fmt.Errorf(
+			"secret foundation tax should non-negative and less than one: %s", p.SecretFoundationTax,
+		)
+	}
```

Comment:
Should we add here a check that either both `SecretFoundationTax` & `SecretFoundationAddress` are set, or both are not set?
Just to avoid an undefined behavior

Reply:
So I can't recall exactly where and how `ValidateBasic` is used on paramsets. Regardless, once params are changed via proposals, they're validated against their _individual_ validation functions, so we can't enforce this mutli-param validation anyway.

So the best thing to do here is make sure the chain starts with valid parameters and whenever param change proposals are made, that both values are explicitly included.